crates/foundry/evm/core/src/backend/mod.rs

```diff
@@ -2046,54 +2198,83 @@ fn is_contract_in_state(journaled_state: &JournaledState, acc: Address) -> bool
 }
 
 /// Updates the env's block with the block's data
-fn update_env_block(env: &mut Env, block: &AnyRpcBlock) {
-    env.block.timestamp = U256::from(block.header.timestamp);
-    env.block.coinbase = block.header.beneficiary;
-    env.block.difficulty = block.header.difficulty;
-    env.block.prevrandao = Some(block.header.mix_hash.unwrap_or_default());
-    env.block.basefee = U256::from(block.header.base_fee_per_gas.unwrap_or_default());
-    env.block.gas_limit = U256::from(block.header.gas_limit);
-    env.block.number = U256::from(block.header.number);
-    if let Some(excess_blob_gas) = block.header.excess_blob_gas {
-        env.block.blob_excess_gas_and_price =
-            Some(BlobExcessGasAndPrice::new(excess_blob_gas, false));
-    }
+fn update_env_block<BlockT, TxT, HardforkT>(
+    env: &mut EvmEnv<BlockT, TxT, HardforkT>,
+    block: &AnyRpcBlock,
+) {
+    todo!()
+    // env.block.timestamp = block.header.timestamp;
+    // env.block.beneficiary = block.header.beneficiary;
+    // env.block.difficulty = block.header.difficulty;
+    // env.block.prevrandao = Some(block.header.mix_hash.unwrap_or_default());
+    // env.block.basefee = block.header.base_fee_per_gas.unwrap_or_default();
+    // env.block.gas_limit = block.header.gas_limit;
+    // env.block.number = block.header.number;
+    // if let Some(excess_blob_gas) = block.header.excess_blob_gas {
+    //     env.block.blob_excess_gas_and_price =
+    //         Some(BlobExcessGasAndPrice::new(excess_blob_gas, false));
+    // }
```

Comment:
Note: EDR doesn't follow this kind of workflow. Rather, we construct take the header from a remote block. The header can be converted into an `revm::BlockEnv`.